### PR TITLE
Added /user api in auth

### PR DESCRIPTION
**Pull request by tusharthakurepc1**

### CODE REVIEW

File: src/routes/external.route.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ class ExternalRoute implements Routes {
   }
 
   private initializeAuthRoutes(prefix: string) {
+    this.router.post(`${prefix}/user`, this.sessionMiddleware.validate, asyncWrapper(this.authenticateController.initialUser));
```
Comment (pratik-edu) [inline src/routes/external.route.ts:39]: Add Comments Here Please.
It should be only `/user` not `/auth/user `

File: src/controllers/authenticate.controller.ts [hunk 2]
```diff
@@ -4,10 +4,31 @@ import { Response, Request } from 'express';
...
   private authenticateService = new AuthenticateService();
+  // Https
+  private pathshalaInternal = new PathshalaInternal();
+
+  public initialUser = async (req: Request, res: Response) => {
+    const domain = req.get('host');
+    if (!domain) {
+      throw new HandledError('Domain not found');
+    }
+    const schoolDetails = await this.pathshalaInternal.getSchoolDetailByDomainName({ domain });
```
Comment (pratik-edu) [inline src/controllers/authenticate.controller.ts:23]: http call on controller layer 🤔??

File: src/controllers/authenticate.controller.ts [hunk 3]
```diff
@@ -4,10 +4,31 @@ import { Response, Request } from 'express';
 import AuthenticateService from '@/services/authenticate.service';
 // Typings
 import { sendOtpRequestBody, userLoginRequestBody, userSignupRequestBody, verifyOtpRequestBody } from './typings/authenticate.controller';
+// Https
+import PathshalaInternal from '@/https/pathshala.http';
+// ErrorHandler
+import { HandledError } from '@/exceptions/HandledError';
```
Comment (pratik-edu) [inline src/controllers/authenticate.controller.ts:10]: exception handling should be service layer

File: src/https/pathshala.http.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { RequestMethodEnum } from '@/constants/enum';
+// Typings
+import { ISchool } from '@/typings/pathshala';
+
+export default class PathshalaInternal {
+  private pathshalaService = services.pathshala_service;
+
+  /**
+   * Retrive all school details by identifier
+   * @param {payload} identifier for fetching school details
```
Comment (HiAkshat) [inline src/https/pathshala.http.ts:15]:
```suggestion
   * @param { Object } payload - identifier for fetching school details
```

File: src/https/pathshala.http.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import fetch from 'node-fetch';
+// Config
+import { services } from '@/config';
+// Constants
+import { RequestMethodEnum } from '@/constants/enum';
+// Typings
+import { ISchool } from '@/typings/pathshala';
+
+export default class PathshalaInternal {
+  private pathshalaService = services.pathshala_service;
```
Comment (HiAkshat) [inline src/https/pathshala.http.ts:11]:
```suggestion
  // Services
  private pathshalaService = services.pathshala_service;
```

File: src/services/authenticate.service.ts [hunk 6]
```diff
@@ -29,6 +31,19 @@ class AuthenticateService {
   private emailHelper = new EmailHelper();
   // Formatters
   private authenticateFormatter = new AuthenticateFormatter();
+  // Https
+  private pathshalaInternal = new PathshalaInternal();
+
+  /**
+   * Fetching school by calling pathshala service
+   * @param payload containing the identifier for fetching the school details
```
Comment (HiAkshat) [inline src/services/authenticate.service.ts:39]:
```suggestion
   * @param  {Object} identifier - containing the identifier for fetching the school details
```

File: src/typings/pathshala.d.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (HiAkshat) [inline src/typings/pathshala.d.ts:1]: remove space

File: src/typings/pathshala.d.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+export interface ISchool {
+  name: string;
+  address: {
+    line1: string;
+    line2?: string;
+    city: string;
+    state: string;
+    postalCode: string;
+  };
+  boardAffiliation: 'CBSE' | 'ICSE' | 'State' | 'International';
```
Comment (HiAkshat) [inline src/typings/pathshala.d.ts:11]: make an enum and import

**Review by HiAkshat — CHANGES_REQUESTED** [review]

**Comment by HiAkshat** [comment]
Run prettify to fix lint issue.

File: src/controllers/authenticate.controller.ts [hunk 9]
```diff
@@ -9,6 +9,28 @@ class AuthenticateController {
...
+   * - Http Call: Calls pathshala for fetching school details
+   * ```
+   * POST: /api/v1/platform/user
+   * ```
+   * @param req - The HTTP request object containing the user's login details.
+   * @param res - The HTTP response object used to send the response back to the client.
+   */
+  public initialUser = async (req: Request, res: Response) => {
+    const domain = req.get('host');
+    const schoolDetails = await this.authenticateService.initialUserRequest({ domain });
```
Comment (pratik-edu) [inline src/controllers/authenticate.controller.ts:24]: is there a need to send it in object.
as ideally one controller should call one service only.
no other controller can call same service directly.
so ATM its fine to send it without object.

File: src/services/authenticate.service.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
 // Modules
-import R from 'ramda';
+import R, { identical } from 'ramda';
```
Comment (pratik-edu) [inline src/services/authenticate.service.ts:2]: why identical??

File: src/services/authenticate.service.ts [hunk 11]
```diff
@@ -29,6 +31,19 @@ class AuthenticateService {
...
+  // Https
+  private pathshalaInternal = new PathshalaInternal();
+
+  /**
+   * Fetching school by calling pathshala service
+   * @param payload containing the identifier for fetching the school details
+   */
+  public initialUserRequest = async (identifier: { domain?: string }) => {
+    if (!identifier.domain) {
+      throw new HandledError('Domain not found');
```
Comment (pratik-edu) [inline src/services/authenticate.service.ts:43]: Add status code here

**Comment by pratik-edu** [comment]
> Run prettify to fix lint issue.

Valid Comment

File: src/routes/external.route.ts [hunk 12]
```diff
@@ -33,6 +33,12 @@ class ExternalRoute implements Routes {
   constructor() {
     this.initializeAuthRoutes(`${this.path}/auth`);
     this.initializeSchoolRoutes(`${this.path}/school`);
```
Comment (pratik-edu) [inline src/routes/external.route.ts:35]: remove or mark all school related api as deprecated.
also update script to call internally to pathshala service for creation of school.

**Review by pratik-edu — APPROVED** [review]

**Review by pratik-edu — APPROVED** [review]

**Review by AbhishekRathore1210 — APPROVED** [review]

**Review by HiAkshat — APPROVED** [review]

**Pull request merged**